lib/util.js

```diff
@@ -140,13 +140,13 @@ const determineModifiedFiles = async function(token, sourcePath) {
         // maximum of 300 files are loaded (page size is 30, max 10 pages)
         let page;
         for(page = 1; page <= MAX_PAGE; page++) {
-            const compareResponse = await octokit.rest.repos.compareCommitsWithBasehead({
+            const commitResponse = await octokit.rest.repos.getCommit({
                 ...context.repo,
-                basehead: `${eventData.before}...${eventData.after}`,
+                ref: eventData.after,
```

Comment:
This change will now only consider the head commit of the commits that have been pushed. If multiple commits have been pushed at once, then the file list won't be complete.

The push event is described here: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push
The sample payload seems to be for a tag, that has been pushed - interestingly it also has `before=00000...`. That means, the same problem (#128) should occur for if a tag has been pushed.

The payload also contains an array `commits[]` - in the example it's empty. I guess this is because it's a tag, that has been pushed. But for usual pushes, that should contain all the commits that are contained in the push. If there is no previous commit (`before=0000....`) we maybe need to loop through the commits and request each commit with your suggested method (`getCommit`).

Also there are other interesting flags in the push event: e.g. if a branch/tag is deleted, we would also get a push event and likely fail to determine the modified files.

Reply:
Thanks for the clarification on the `push` event payload. I misunderstood and thought the `before` was simply the immediate prior commit to what `after` represented. But it seems GitHub webhooks may be trying to be efficient by collapsing multiple commits into a single payload.

I'll reconsider my approach for handling the specific instance when `before=0000....` per your recommendations.